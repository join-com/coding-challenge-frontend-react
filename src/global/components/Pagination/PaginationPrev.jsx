import React from 'react';
import PropTypes from 'prop-types';

import I18n from '../../../utils/I18n';

import Button from '../Button';

import './css/PaginationPrev.css';

const PaginationPrev = ({
  totalPages, pageNumber, onPageNumberChange,
}) => {
  if (totalPages > 1 && pageNumber > 1) {
    return (
      <Button
        className="PaginationPrev__button"
        text={I18n.t('en.pagination.prev')}
        params={{ pageNumber: pageNumber - 1 }}
        onClick={onPageNumberChange}
      />
    );
  }
  return null;
};

export default PaginationPrev;

PaginationPrev.propTypes = {
  totalPages: PropTypes.number,
  pageNumber: PropTypes.number,
  onPageNumberChange: PropTypes.func,
};

PaginationPrev.defaultProps = {
  totalPages: 0,
  pageNumber: 1,
  onPageNumberChange: () => {},
};
