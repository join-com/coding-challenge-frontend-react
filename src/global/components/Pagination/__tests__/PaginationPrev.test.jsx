import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import PaginationPrev from '../PaginationPrev';

const props = {
  totalPages: 3,
  pageNumber: 2,
};

describe('PaginationPrev', () => {
  it('should render the component for totalPages > 1 & pageNumber > 1', () => {
    const tree = shallow(<PaginationPrev {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should not render the component for pageNumber = 1', () => {
    const tree = shallow(<PaginationPrev {...props} pageNumber={1} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should not render the component for totalPages = 1', () => {
    const tree = shallow(<PaginationPrev {...props} totalPages={1} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
