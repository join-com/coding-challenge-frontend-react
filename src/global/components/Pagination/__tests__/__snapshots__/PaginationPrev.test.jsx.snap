// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationPrev should not render the component for pageNumber = 1 1`] = `""`;

exports[`PaginationPrev should not render the component for totalPages = 1 1`] = `""`;

exports[`PaginationPrev should render the component for totalPages > 1 & pageNumber > 1 1`] = `
<ButtonWrapper
  className="PaginationPrev__button"
  onClick={[Function]}
  params={
    Object {
      "pageNumber": 1,
    }
  }
  text="Prev"
/>
`;
